### PR TITLE
Handle importing of different files

### PR DESCRIPTION
**Pull request by microbit-grace**
When importing non-MakeCode hex, or invalid json files, or files with unexpected file extensions (e.g. .png), show import error dialog.
Screenshot 2024-10-30 at 10 11 47

When importing non-CreateAI  (has no ML extension) MakeCode hex files, show dialog telling users how to view their project.
Screenshot 2024-10-30 at 10 13 18

### CODE REVIEW

**Comment by microbit-grace** [comment]
What about when user imports a CreateAI project without data samples? Should we pop-up the same dialog as the non-createAI hex dialog?

File: src/hooks/project-hooks.tsx [hunk 1]
```diff
@@ -178,16 +184,25 @@ export const ProjectProvider = ({
...
         }
       } else if (fileExtension === "hex") {
-        driverRef.current!.importFile({
-          filename: file.name,
-          parts: [await readFileAsText(file)],
-        });
+        const hex = await readFileAsText(file);
+        const makeCodeMagicMark = "41140E2FB82FA2BB";
+        // Check if is a MakeCode hex, otherwise show error dialog.
+        if (hex.includes(makeCodeMagicMark)) {
```
Comment (microbit-grace) [inline src/hooks/project-hooks.tsx:193]: Logic taken from https://github.com/microbit-foundation/microbit-android/blob/d1a49f745d1917d3d4e8de519e4deba9aa62c950/app/src/main/java/com/samsung/microbit/ui/activity/ProjectActivity.java#L2583-L2590

Comment (microbit-grace) [reply src/hooks/project-hooks.tsx:193]: Not sure if I should leave it as a comment in the actual code 🤔

**Comment by microbit-grace** [comment]
> What about when user imports a CreateAI project without data samples? Should we pop-up the same dialog as the non-createAI hex dialog?

Discussed, agreed that users would probably have used the tool by this point and would know what to do if they import a CreateAI project so no pop-up necessary for this case.

**Comment by microbit-robert** [comment]
Maybe this text "Train a model to view your code." should also include a reference to recording data samples as we know the user won't have any when this dialog shows.

**Comment by microbit-robert** [comment]
Is this too friendly and easy to ignore? One to discuss with team.
image

File: src/hooks/project-hooks.tsx (hunk 1 above)
Comment (microbit-matt-hillsdon) [reply src/hooks/project-hooks.tsx:193]: No that's fine.

**Pull request merged**